### PR TITLE
VPN-5251 - Declare metrics

### PR DESCRIPTION
**Pull request by brizental**
Decided to pull this out of the original VPN-5251 (https://github.com/mozilla-mobile/mozilla-vpn-client/pull/8139) PR, just because it adds a lot of noise.

### CODE REVIEW

**Comment by brizental** [comment]
!! Reminder: it is your responsibility to complete and check the correctness of
!! this automatically-generated request skeleton before requesting Data
!! Collection Review. See https://wiki.mozilla.org/Data_Collection for details.

DATA REVIEW REQUEST
1. What questions will you answer with this data?

-   **P1** As a Product Manager, I want to understand the funnel efficiency of the account registration process (per device), understanding drop off rates in every screen.

-   **P1** As a Product Manager, I want to understand the funnel efficiency of the subscription creation process (per device), understanding drop off rates in every screen.

-   **P2** As a Product Manager, I want to know on which screens users are feeling lost and click "Get Help", which could be an indication of missing / inaccurate / confusing communication.

-   **P2** As a Product Manager, I want to understand how often users are having trouble with the 6-digit verification code (success/failure events) and request a new one.

-   **P3** As a Product Manager, I want to understand when users are interested in the Terms of Service and Privacy Policy. We have an abundance of those links and knowing when they are not being used could lead to removing some noise from the UI.

-   **P3** As a Product Manager, I want to be able to understand how often people use the pre-login carrousel.

-   **P4** As a Product Manager, I want to know if users are using the 'paste' button in the verification screen.

2. Why does Mozilla need to answer these questions? Are there benefits for users?
   Do we need this information to address product or business requirements?

This data is necessary for the Mozilla VPN product team to understand the 
behaviour of the authentication and subscription features and how
to make that flow better fol users.

3. What alternative methods did you consider to answer these questions?
   Why were they not sufficient?

None.

4. Can current instrumentation answer these questions?

Partially only.

5. List all proposed measurements and indicate the category of data collection for each
   measurement, using the Firefox data collection categories found on the Mozilla wiki.

Measurement Name | Measurement Description | Data Collection Category | Tracking Bug
---------------- | ----------------------- | ------------------------ | ------------
`impression.signup_screen` | The user has entered the "Sign up" screen.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`impression.enter_email_screen` | The user has entered the "Enter email" screen.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`impression.create_password_screen` | The user has entered the "Create password" screen.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`impression.enter_verification_code_screen` | The user has entered the screen to enter email verification code.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`impression.enter_password_screen` | The user has entered the "Enter password" screen.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`impression.enter_security_code_screen` | The user has entered the screen to enter a TOTP security code.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`impression.subscription_needed_screen` | The user has entered the "Subscription needed" screen.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`impression.confirming_subscription_screen` | The user has entered the "Confirming subscription" screen.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`impression.continue_in_browser_screen` | The user has entered the "Continue in browser" screen.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`interaction.get_help_selected` | The "Get help" link was clicked.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`interaction.signup_selected` | The "Sign up" button was clicked.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`interaction.already_a_subscriber` | The "Already a subscriber" button was clicked.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`interaction.back_selected` | The back arrow button was clicked.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`interaction.continue_selected` | The "Continue" button was clicked in one of the auth screens.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`interaction.change_email_selected` | The "Change email" button was clicked.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`interaction.paste_password_selected` | The "Paste" button was clicked on the password input.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`interaction.create_account_selected` | The "Create account" button was clicked.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`interaction.privacy_notice_selected` | The "Privacy notice" button was clicked.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`interaction.terms_of_service_selected` | The "Terms of Service" button was clicked.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`interaction.close_selected` | The "Close" button was clicked. This may happen in multiple places in code.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`interaction.verify_selected` | The "Verify" button was clicked in one of the security code verification screens.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`interaction.resend_code_selected` | The "Resend code" button was clicked in one of the security code verification screens.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`interaction.cancel_selected` | A "cancel" button was clicked.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`interaction.sign_in_selected` | A "Sign in" button was clicked.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`interaction.forgot_your_password_selected` | The "Forgot your password?" button was clicked.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`interaction.subscribe_now_selected` | The "Subscribe now" button was clicked.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`interaction.sign_out_selected` | The "Sign out" button was clicked.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`outcome.login_started` | The user has started the login process.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`outcome.registration_started` | The user has started the registration process.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`outcome.two_fa_verification_failed` | The user has just failed the 2FA process.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`outcome.two_fa_verification_succeeded` | The user has just succeeded in the 2FA process.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`outcome.login_ended` | The user has just completed the login process.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`outcome.registration_completed` | The user has just completed the registration process.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`outcome.subscription_started` | The user has just started the subscription process.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251
`outcome.subscription_completed` | The user has just completed the subscription process.  | interaction | https://mozilla-hub.atlassian.net/browse/VPN-5251


6. Please provide a link to the documentation for this data collection which
   describes the ultimate data set in a public, complete, and accurate way.

This collection is Glean so is documented
[in the Glean Dictionary](https://dictionary.telemetry.mozilla.org).

7. How long will this data be collected?

This collection will be collected permanently.
brizental@mozilla.com, vpn-telemetry@mozilla.com will be responsible for the permanent collections.

8. What populations will you measure?

All channels, countries, and locales. No filters.

9. If this data collection is default on, what is the opt-out mechanism for users?

These collections are Glean. The opt-out can be found in the product's preferences.

10. Please provide a general description of how you will analyze this data.

Looker dashboards and explores.

11. Where do you intend to share the results of your analysis?

Within Mozilla.

12. Is there a third-party tool (i.e. not Telemetry) that you
    are proposing to use for this data collection?

No.

**Comment by brizental** [comment]
Converting to draft just until data review is done.

**Comment by travis79** [comment]
> !! Reminder: it is your responsibility to complete and check the correctness of !! this automatically-generated request skeleton before requesting Data !! Collection Review. See [wiki.mozilla.org/Data_Collection](https://wiki.mozilla.org/Data_Collection) for details.
> 
> DATA REVIEW REQUEST
> 
>     1. What questions will you answer with this data?
> 
> 
>     * **P1** As a Product Manager, I want to understand the funnel efficiency of the account registration process (per device), understanding drop off rates in every screen.
> 
>     * **P1** As a Product Manager, I want to understand the funnel efficiency of the subscription creation process (per device), understanding drop off rates in every screen.
> 
>     * **P2** As a Product Manager, I want to know on which screens users are feeling lost and click "Get Help", which could be an indication of missing / inaccurate / confusing communication.
> 
>     * **P2** As a Product Manager, I want to understand how often users are having trouble with the 6-digit verification code (success/failure events) and request a new one.
> 
>     * **P3** As a Product Manager, I want to understand when users are interested in the Terms of Service and Privacy Policy. We have an abundance of those links and knowing when they are not being used could lead to removing some noise from the UI.
> 
>     * **P3** As a Product Manager, I want to be able to understand how often people use the pre-login carrousel.
> 
>     * **P4** As a Product Manager, I want to know if users are using the 'paste' button in the verification screen.
> 
> 
>     2. Why does Mozilla need to answer these questions? Are there benefits for users?
>        Do we need this information to address product or business requirements?
> 
> 
> This data is necessary for the Mozilla VPN product team to understand the behaviour of the authentication and subscription features and how to make that flow better fol users.
> 
>     3. What alternative methods did you consider to answer these questions?
>        Why were they not sufficient?
> 
> 
> None.
> 
>     4. Can current instrumentation answer these questions?
> 
> 
> Partially only.
> 
>     5. List all proposed measurements and indicate the category of data collection for each
>        measurement, using the Firefox data collection categories found on the Mozilla wiki.
> 
> 
> Measurement Name 	Measurement Description 	Data Collection Category 	Tracking Bug
> `impression.signup_screen` 	The user has entered the "Sign up" screen. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `impression.enter_email_screen` 	The user has entered the "Enter email" screen. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `impression.create_password_screen` 	The user has entered the "Create password" screen. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `impression.enter_verification_code_screen` 	The user has entered the screen to enter email verification code. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `impression.enter_password_screen` 	The user has entered the "Enter password" screen. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `impression.enter_security_code_screen` 	The user has entered the screen to enter a TOTP security code. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `impression.subscription_needed_screen` 	The user has entered the "Subscription needed" screen. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `impression.confirming_subscription_screen` 	The user has entered the "Confirming subscription" screen. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `impression.continue_in_browser_screen` 	The user has entered the "Continue in browser" screen. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `interaction.get_help_selected` 	The "Get help" link was clicked. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `interaction.signup_selected` 	The "Sign up" button was clicked. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `interaction.already_a_subscriber` 	The "Already a subscriber" button was clicked. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `interaction.back_selected` 	The back arrow button was clicked. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `interaction.continue_selected` 	The "Continue" button was clicked in one of the auth screens. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `interaction.change_email_selected` 	The "Change email" button was clicked. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `interaction.paste_password_selected` 	The "Paste" button was clicked on the password input. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `interaction.create_account_selected` 	The "Create account" button was clicked. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `interaction.privacy_notice_selected` 	The "Privacy notice" button was clicked. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `interaction.terms_of_service_selected` 	The "Terms of Service" button was clicked. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `interaction.close_selected` 	The "Close" button was clicked. This may happen in multiple places in code. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `interaction.verify_selected` 	The "Verify" button was clicked in one of the security code verification screens. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `interaction.resend_code_selected` 	The "Resend code" button was clicked in one of the security code verification screens. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `interaction.cancel_selected` 	A "cancel" button was clicked. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `interaction.sign_in_selected` 	A "Sign in" button was clicked. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `interaction.forgot_your_password_selected` 	The "Forgot your password?" button was clicked. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `interaction.subscribe_now_selected` 	The "Subscribe now" button was clicked. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `interaction.sign_out_selected` 	The "Sign out" button was clicked. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `outcome.login_started` 	The user has started the login process. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `outcome.registration_started` 	The user has started the registration process. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `outcome.two_fa_verification_failed` 	The user has just failed the 2FA process. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `outcome.two_fa_verification_succeeded` 	The user has just succeeded in the 2FA process. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `outcome.login_ended` 	The user has just completed the login process. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `outcome.registration_completed` 	The user has just completed the registration process. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `outcome.subscription_started` 	The user has just started the subscription process. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> `outcome.subscription_completed` 	The user has just completed the subscription process. 	interaction 	[mozilla-hub.atlassian.net/browse/VPN-5251](https://mozilla-hub.atlassian.net/browse/VPN-5251)
> 
>     6. Please provide a link to the documentation for this data collection which
>        describes the ultimate data set in a public, complete, and accurate way.
> 
> 
> This collection is Glean so is documented [in the Glean Dictionary](https://dictionary.telemetry.mozilla.org).
> 
>     7. How long will this data be collected?
> 
> 
> This collection will be collected permanently. [brizental@mozilla.com](mailto:brizental@mozilla.com), [vpn-telemetry@mozilla.com](mailto:vpn-telemetry@mozilla.com) will be responsible for the permanent collections.
> 
>     8. What populations will you measure?
> 
> 
> All channels, countries, and locales. No filters.
> 
>     9. If this data collection is default on, what is the opt-out mechanism for users?
> 
> 
> These collections are Glean. The opt-out can be found in the product's preferences.
> 
>     10. Please provide a general description of how you will analyze this data.
> 
> 
> Looker dashboards and explores.
> 
>     11. Where do you intend to share the results of your analysis?
> 
> 
> Within Mozilla.
> 
>     12. Is there a third-party tool (i.e. not Telemetry) that you
>         are proposing to use for this data collection?
> 
> 
> No.

# Data Review

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way?

Yes, through the metrics.yaml file and the [Glean Dictionary](https://dictionary.telemetry.mozilla.org/).

2) Is there a control mechanism that allows the user to turn the data collection on and off?

Yes, through the data preferences in the application settings.

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

Permanent collection to be monitored by brizental and the VPN team

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Category 2, Interaction data

5) Is the data collection request for default-on or default-off?

Default-on

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?

No

7) Is the data collection covered by the existing Firefox privacy notice?

Yes

8) Does the data collection use a third-party collection tool?

No

## Result

data-review+

File: src/telemetry/metrics.yaml [hunk 1]
```diff
@@ -725,7 +1455,7 @@ sample:
     bugs:
       - https://mozilla-hub.atlassian.net/browse/VPN-3743
     data_reviews:
-      - TODO
+      - https://github.com/mozilla-mobile/mozilla-vpn-client/pull/8411#issuecomment-1783192931
```
Comment (brizental) [inline src/telemetry/metrics.yaml:1458]: Oh, this is a problem.

Comment (brizental) [reply src/telemetry/metrics.yaml:1458]:
```suggestion
      - TODO
```

**Review by Gela — APPROVED** [review]

**Pull request merged**